s3-source-connector/src/integration-test/java/io/aiven/kafka/connect/s3/source/IntegrationTest.java

```diff
@@ -205,12 +205,15 @@ void bytesTestBasedOnMaxMessageBytes(final TestInfo testInfo) throws IOException
         // Verify that the correct data is read from the S3 bucket and pushed to Kafka
         assertThat(records).containsExactly("AA", "BB", "CC", "DD", "EE");
 
+        // Wait until all the events to connect-offsets topic are produced
+        Thread.sleep(10_000);
         // Verify offset positions
-        verifyOffsetPositions(offsetKeys, 1);
+        final Map<String, Object> offsetKeyPositions = verifyOffsetPositions(offsetKeys, 1);
+        offsetKeyPositions.forEach((key, value) -> assertThat(value).isEqualTo(5));
```

Comment:
```suggestion
        assertThat(offsetKeyPositions).values().containsOnly(5);
```

Reply:
Here I need to asssert on the value of position, and not the size of values.